### PR TITLE
do not crash for submenu with double quotes and also filter out snapp…

### PR DESCRIPTION
**Pull request by jreidinger**
…er menuentry

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,4 +1,11 @@
 -------------------------------------------------------------------
+Wed Jul 13 07:40:52 UTC 2016 - jreidinger@suse.com
+
+- Do not crash for submenu with double quotes (bnc#986005)
+- filter out snapper menuentry ( using double quotes )
```
Comment (mvidner) [inline CHANGELOG]: why should a generic grub config be interested in snapper? Did you mean to say this instead?
```
- Handle submenus with double quotes (do not crash)
- Filter out menuentries with double quotes (used by snapper in opensuse/SLE)
```

Comment (jreidinger) [inline CHANGELOG]: question is if we should document reality or contract. For me change contract is to not crash on double quotes and filter out snapper menu entry. If someone report that their distro use double quotes, I am willing to change code to fix it and really only filter out snapper, still considering it only bug fix and not added feature. So filtering out double quotes is not set in stones nor feature on which someone should rely.

Comment (mvidner) [inline CHANGELOG]:
```
- Do not crash for submenu with double quotes (bnc#986005)
- Filter out non-bootable menuentries (a Snapper help label, using double quotes )
```

**Comment by mvidner** [comment]
continuing the yard comment for boot_entries:
```
# Some entries are not in fact bootable, such as the "run snaper rollback" hint-only entry on SUSE.
# They are ignored. As a hack, they are recognized by double quote delimiters while the regular
# entries use single quotes.
```

File: spec/grub_cfg_spec.rb [hunk 2]
```diff
@@ -52,6 +52,35 @@
...
       )
     end
+
+    context "grub.cfg with snapper boot entry" do
+      let(:memory_file) do
+        path = File.expand_path("../fixtures/grub-with-snapper.cfg", __FILE__)
+        CFA::MemoryFile.new(File.read(path))
+      end
+
+      it "filter out snapper entry" do
```
Comment (mvidner) [inline spec/grub_cfg_spec.rb]:
```rb
it "filters out unbootable entries" do
```

**Comment by mvidner** [comment]
Thank you, LGTM!

**Pull request merged**